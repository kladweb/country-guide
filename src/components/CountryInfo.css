/*.content {*/
/*  position: relative;*/
/*}*/


.country-info {
  /*margin: 0 auto;*/
  width: 100%;
  position: fixed;
  left: 0;
  bottom: 0;
  color: #eaeaea;
  /*text-align: center;*/
  background: #7a4d40;
}

.country-info__content {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.country-info h2,
.country-info h3 {
  font-weight: 400;
}

.sectionInfo {
  display: inline-block;
  /*vertical-align: middle;*/
}

.flag-preview-info {
  height: 4rem;
  width: 20%;
  margin: 1rem;
  object-fit: contain;
}


