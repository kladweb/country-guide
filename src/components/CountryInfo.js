import React from 'react';

export const CountryInfo = ({ fio, balance } ) => {

  return (
    <h1>
      клиент &laquo;{fio}&raquo;, баланс {balance}
    </h1>
  );

};
