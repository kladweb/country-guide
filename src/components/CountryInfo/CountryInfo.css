.country-info {
  width: 100%;
  padding: 0;
  position: fixed;
  left: 0;
  bottom: 0;
  color: #eaeaea;
  background: #7a4d40;
  border-top: solid 5px #939999;
  border-bottom: solid 5px #939999;
}

.show {
  animation: show-info 500ms ease-in-out;
}

@keyframes show-info {
  from {
    transform: translateY(100%);
  }
  to {
    transform: translateY(0);
  }
}

.hide {
  animation: hide-info 500ms ease-in-out;
}

@keyframes hide-info {
  from {
    transform: translateY(0);
  }
  to {
    transform: translateY(100%);
  }
}

.hideAll {
  transform: translateY(100%);
}

.country-info__content {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.country-info h2,
.country-info h3 {
  font-weight: 400;
}

.country-info h3 {
  padding-left: 20%;
  font-weight: 400;
}

.country-info .country-info__name {
  padding-bottom: 0.5rem;
  font-size: 2rem;
  text-align: center;
}

.country-info__properties {
  font-size: 1.25rem;
}

.country-info__value {
  font-size: 1.5rem;
  font-weight: 500;
  text-align: left;
}

.sectionInfo {
  display: inline-block;
  width: 33.3%;
}

.flag-preview-info {
  height: 4rem;
  width: 20%;
  margin: 1rem;
  object-fit: contain;
}

.button-close {
  position: absolute;
  font-size: 1rem;
  right: 0.75rem;
  top: 0.5rem;
  cursor: pointer;
  border: solid 1px;
  border-radius: 50%;
}

@media (max-width: 1080px) {
  .country-info .country-info__name {
    font-size: 1.5rem;
  }

  .country-info__properties {
    width: 70%;
    vertical-align: middle;
  }

  .country-info__properties {
    font-size: 1rem;
  }

  .country-info__value {
    font-size: 1.25rem;
  }

  .country-info h3 {
    display: inline-block;
    padding: 0.25rem 2rem 0.7rem;
  }

  .flag-preview-info {
    display: none;
  }
}

@media (max-width: 600px) and (orientation: portrait) {
  .country-info__content {
    display: flex;
    flex-direction: column;
  }

  .country-info__name {
    width: auto;
  }

  .country-info h3 {
    padding: 0.25rem 0;
  }

  .country-info__properties {
    padding-bottom: 0.9rem;
    text-align: center;
  }
}

@media (max-width: 360px) and (orientation: portrait) {
  .country-info__properties {
    font-size: 0.75rem;
  }

  .country-info__value {
    font-size: 0.9rem;
  }
}