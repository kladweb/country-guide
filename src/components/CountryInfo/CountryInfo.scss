@import "../../styles/variables.scss";

.country-info {
  width: 100%;
  padding: 0;
  position: fixed;
  left: 0;
  bottom: 0;
  color: $primary-color;
  background: $countries-links;
  border-top: solid 5px $color-light;
  border-bottom: solid 5px $color-light;
}

.show {
  animation: show-info 500ms ease-in-out;
}

@keyframes show-info {
  from {
    transform: translateY(100%);
  }
  to {
    transform: translateY(0);
  }
}

.hide {
  animation: hide-info 500ms ease-in-out;
}

@keyframes hide-info {
  from {
    transform: translateY(0);
  }
  to {
    transform: translateY(100%);
  }
}

.hideAll {
  transform: translateY(100%);
}

.country-info__content {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.sectionInfo {
  margin-top: -0.5rem;
  width: 33.3%;
  display: flex;
  justify-content: center;

  .sectionName {
    text-align: center;
  }

  h2, h3, h4 {
    font-size: 1.5em;
    font-weight: 400;
  }

  .country-info__name {
    font-size: 2em;
  }
}

.country-info__value, .country-info__capital {
  font-weight: 500;
}

.flag-preview-info {
  height: 4rem;
  width: 20%;
  margin: 1rem;
  object-fit: contain;
}

.button-close {
  position: absolute;
  font-size: 1rem;
  right: 0.75rem;
  top: 0.5rem;
  cursor: pointer;
  border: solid 1px;
  border-radius: 50%;
}

@media (max-width: 1080px) {
  .flag-preview-info {
    display: none;
  }

  .sectionInfo {
    margin: 0.25em 0 0.5em;
    width: 50%;
    font-size: 0.9rem;
  }
}

@media (max-width: 600px) and (orientation: portrait) {
  .country-info__content {
    display: flex;
    flex-direction: column;
  }

  .sectionInfo {
    width: 95%;
  }
}

@media (max-width: 360px) and (orientation: portrait) {
  .sectionInfo {
    font-size: 0.8rem;
  }
}
