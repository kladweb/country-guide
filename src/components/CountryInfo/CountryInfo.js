import React, { useEffect, useState } from 'react';
import { useNavigate, useParams } from "react-router-dom";
import { useSelector } from "react-redux";
import './CountryInfo.css';

export const CountryInfo = ({code, name, population, area, pageCount}) => {
  const params = useParams();
  const page = params.part;
  const navigate = useNavigate();
  const isOpenInfoBar = useSelector(state => state.openInfoBar.isOpenInfoBar);
  //делаем отступы между тысячными:
  const people = population.toString().replace(/(\d)(?=(\d{3})+$)/g, '$1 ');
  const square = area.toString().replace(/(\d)(?=(\d{3})+$)/g, '$1 ');

  const [classInfo, changeClassInfo] = useState('country-info show');

  const closeInfo = () => {
    changeClassInfo('country-info hide');
    setTimeout(() => {
      (page) ? navigate(`/${pageCount}/${page}`) : navigate(`/${pageCount}`);
    }, 450);
  }

  // console.log(isOpenInfoBar);
  useEffect(() => {
    if (isOpenInfoBar === 'close') {
      closeInfo();
    }
  }, [isOpenInfoBar]);

  return (
    <div className={classInfo}>
      <div className='content country-info__content'>
        <h2 className='sectionInfo country-info__name'>&laquo;{name}&raquo;</h2>
        <img className='sectionInfo flag-preview-info' src={`/img/flags/${code}.png`} alt={name} />
        <div className='sectionInfo country-info__properties'>
          <h3>Population: <span className='country-info__value'>{people}</span></h3>
          <h3>Total area: <span className='country-info__value'>{square}</span> km<sup>2</sup></h3>
        </div>
      </div>
      <div className='material-icons button-close' onClick={closeInfo}>close</div>
    </div>
  );

};