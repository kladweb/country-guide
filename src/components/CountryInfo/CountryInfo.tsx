import React, { useEffect, useState } from 'react';
import { useNavigate, useParams } from "react-router-dom";
import './CountryInfo.scss';
import { ICountries } from "../../types/globalTypes";
import { useAppSelector } from "../../redux/store";

export const CountryInfo: React.FC<ICountries> = ({code, name, capital, population, area}) => {
  const params = useParams();
  const page = params.part;
  const navigate = useNavigate();
  const isOpenInfoBar = useAppSelector(state => state.openInfoBar.isOpenInfoBar);
  //make indents between thousandths:
  const people = population.toString().replace(/(\d)(?=(\d{3})+$)/g, '$1 ');
  const square = area.toString().replace(/(\d)(?=(\d{3})+$)/g, '$1 ');

  const [classInfo, changeClassInfo] = useState('country-info show');

  const closeInfo = () => {
    changeClassInfo('country-info hide');
    setTimeout(() => {
      navigate(`/countries/${page}`);
    }, 450);
  }

  useEffect(() => {
    if (isOpenInfoBar === 'close') {
      closeInfo();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isOpenInfoBar]);

  return (
    <div className={classInfo}>
      <div className='content country-info__content'>
        <div className='sectionInfo'>
          <div className='sectionName'>
            <h2 className='country-info__name'>{name}</h2>
            <h4>Capital: <span className='country-info__capital'>{capital}</span></h4>
          </div>
        </div>
        <img className='sectionInfo flag-preview-info' src={`/img/flags/${code}.png`} alt={name}/>
        <div className='sectionInfo'>
          <div className='country-info__properties'>
            <h3>Population: <span className='country-info__value'>{people}</span></h3>
            <h3>Total area: <span className='country-info__value'>{square}</span> km<sup>2</sup></h3>
          </div>
        </div>
      </div>
      <div className='material-icons button-close' onClick={closeInfo}>close</div>
    </div>
  );
};
