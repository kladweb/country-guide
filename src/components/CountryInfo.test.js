import React from 'react';

import renderer from 'react-test-renderer';

import { CountryInfo } from "./CountryInfo";

test('работа CountryInfo', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <CountryInfo
      code={100}
      name={'Country'}
      population={1000}
      area={5000}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

});