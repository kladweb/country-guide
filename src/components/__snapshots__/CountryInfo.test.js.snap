// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа CountryInfo 1`] = `
<div
  className="country-info show"
>
  <div
    className="content country-info__content"
  >
    <h2
      className="sectionInfo country-info__name"
    >
      «
      Country
      »
    </h2>
    <img
      alt="Country"
      className="sectionInfo flag-preview-info"
      src="/img/flags/100.png"
    />
    <div
      className="sectionInfo country-info__properties"
    >
      <h3>
        Population: 
        <span
          className="country-info__value"
        >
          1 000
        </span>
      </h3>
      <h3>
        Total area: 
        <span
          className="country-info__value"
        >
          5 000
        </span>
         km
        <sup>
          2
        </sup>
      </h3>
    </div>
  </div>
</div>
`;
